galsim/scene.py

```diff
@@ -0,0 +1,425 @@
+
+            # Find the file.
+            use_file_name, _, _ = \
+                _parse_files_dirs(file_name, image_dir, dir, noise_dir)
+
+            # Read in data.
+            cat = pyfits.getdata(use_file_name)
+
+            # If requested, select galaxies based on existence of a usable fit.
+            if exclude_fail:
+                sersicfit_status = cat['fit_status'][:,4]
+                bulgefit_status = cat['fit_status'][:,0]
+                use_fit_ind = np.where(
+                    (sersicfit_status > 0) &
+                    (sersicfit_status < 5) &
+                    (bulgefit_status > 0) &
+                    (bulgefit_status < 5)
+                    )[0]
+                self.cat = cat[use_fit_ind]
+            else:
+                self.cat = cat
+            self.obj_type = 'parametric'
+            self.nobjects = len(self.cat)
+
+    def makeObj(self, indices=None, chromatic=False, pad_size=None, rng=None):
+        """
+        Routine to construct GSObjects corresponding to catalog entries with particular indices.
+
+        The fluxes are set such that drawing into an image with the COSMOS bandpass and pixel scale
+        should give the right pixel values to mimic the actual COSMOS image.
+
+        @param indices    The indices of the catalog entries for which GSObjects should be
+                          constructed. This should be either a single number or an iterable.  If
+                          None, then a single galaxy is chosen at random.
+                          [default: None]
+        @param chromatic  Make this a chromatic object, or not?  [default: False]
+                          It is important to bear in mind that we do not actually have
+                          spatially-resolved color information for these galaxies, so this keyword
+                          can only be True if we are using parametric galaxies.  Even then, we
+                          simply do the most arbitrary thing possible, which is to assign bulges an
+                          elliptical SED, disks a disk-like SED, and Sersic galaxies with
+                          intermediate values of n some intermediate SED.  We assume that the
+                          photometric redshift is the correct redshift for these galaxies (which is
+                          a good assumption for COSMOS 30-band photo-z for these bright galaxies).
+                          For the given SED and redshift, we then normalize to give the right
+                          (observed) flux in F814W.  Note that for a mock "deep" sample, the
+                          redshift distributions of the galaxies would be modified, which is not
+                          included here.
+        @param pad_size   For realistic galaxies, the size of region requiring noise padding, in
+                          arcsec.  If None, then a region that is 0.25 arcsec in size is used.
+                          [default: None]
```

Comment:
Yeah, a larger default makes sense.  To be honest the best option depends a lot on the target PSF (if you're resimulating to make a ground-based image you usually need a larger ```pad_size``` than if you're just degrading it a bit) but if we're just aiming for something semi-reasonable as a default then ~2-ish arcsec would probably be fine.